diff -up redhat-rpm-config-9.0.3/brp-strip.orig redhat-rpm-config-9.0.3/brp-strip
--- redhat-rpm-config-9.0.3/brp-strip.orig	2016-03-29 17:04:57.115353818 +0200
+++ redhat-rpm-config-9.0.3/brp-strip	2016-03-29 17:05:41.232747553 +0200
@@ -9,6 +9,6 @@ fi
 for f in `find $RPM_BUILD_ROOT -type f \( -perm -0100 -or -perm -0010 -or -perm -0001 \) -exec file {} \; | \
         grep -v "^${RPM_BUILD_ROOT}/\?usr/lib/debug"  | \
 	grep -v ' shared object,' | \
-	sed -n -e 's/^\(.*\):[ 	]*ELF.*, not stripped/\1/p'`; do
+	sed -n -e 's/^\(.*\):[ 	]*ELF.*, not stripped.*/\1/p'`; do
 	$STRIP -g "$f" || :
 done
diff -up redhat-rpm-config-9.0.3/brp-strip-shared.orig redhat-rpm-config-9.0.3/brp-strip-shared
--- redhat-rpm-config-9.0.3/brp-strip-shared.orig	2016-03-29 17:05:08.155202106 +0200
+++ redhat-rpm-config-9.0.3/brp-strip-shared	2016-03-29 17:05:58.504510204 +0200
@@ -15,6 +15,6 @@ fi
 for f in `find $RPM_BUILD_ROOT -type f -a -exec file {} \; | \
         grep -v "^${RPM_BUILD_ROOT}/\?usr/lib/debug"  | \
 	grep ' shared object,' | \
-	sed -n -e 's/^\(.*\):[ 	]*ELF.*, not stripped/\1/p'`; do
+	sed -n -e 's/^\(.*\):[ 	]*ELF.*, not stripped.*/\1/p'`; do
 	$STRIP --strip-unneeded "$f"
 done
