--- redhat-rpm-config-9.0.3/brp-java-repack-jars.orig	2016-02-09 16:49:40.253986272 +0100
+++ redhat-rpm-config-9.0.3/brp-java-repack-jars	2016-02-09 16:51:42.065412746 +0100
@@ -26,6 +26,8 @@
 	# unpack every jar, set the date of the files and directories and 
 	# repack the jar
 	for j in $JARS ; do
+		JAROWN=`ls -l $j | cut -d' ' -f3`
+		JARGRP=`ls -l $j | cut -d' ' -f4`
 		JARNAME=`basename $j`
 		JTMPDIR=`mktemp -d -p $RPM_BUILD_ROOT/tmp $JARNAME.tmpdir.XXXXXXXXXX` || exit 1
 		JARDIR=`mktemp -d -p $RPM_BUILD_ROOT/tmp $JARNAME.jardir.XXXXXXXXXX` || exit 1
@@ -84,6 +86,9 @@
 		fi
 		popd > /dev/null
 
+		chown $JAROWN $j
+		chgrp $JARGRP $j
+
 		# Cleanup.
 		rm -rf $JTMPDIR
 		rm -rf $JARDIR
